docs/reference/search/search-your-data/retrieve-selected-fields.asciidoc

```diff
@@ -325,6 +325,77 @@ will return only the users first name but still maintain the structure of the ne
 However, when the `fields` pattern targets the nested `user` field directly, no
 values will be returned since the pattern doesn't match any leaf fields.
 
+[discrete]
+[[search-fields-flattened]]
+==== Handling of flattened fields
```

Comment:
It would be nice to move this to the `flattened` field type documentation. That would help keep this high-level documentation streamlined and easy to approach.